spec/models/solidus_subscriptions/subscription_spec.rb

```diff
@@ -14,6 +14,16 @@
       with_message('is not a valid currency code')
   end
 
+  it 'validates payment_source ownership' do
+    subscription = create(:subscription)
+
+    subscription.update(payment_source: create(:credit_card))
+    expect(subscription.errors.messages[:payment_source]).to include('does not belong to the user associated with the subscription')
+
+    subscription.update(payment_source: create(:credit_card, user: subscription.user))
+    expect(subscription.errors.messages[:payment_source]).not_to include('does not belong to the user associated with the subscription')
```

Comment:
What do you think of extracting it to a different example? We would have only one behavior for each example in this case.

Reply:
I agree this would be a good idea, although given it's a small change I'd consider it non-blocking.